modules/aggregation-layers/src/cpu-grid-layer/grid-aggregator.js

```diff
@@ -92,11 +94,17 @@ function pointsToGridHashing(props, aggregationParams) {
     if (Number.isFinite(x) && Number.isFinite(y)) {
       const yIndex = Math.floor((y + offsets[1]) / gridOffset.yOffset);
       const xIndex = Math.floor((x + offsets[0]) / gridOffset.xOffset);
-      const key = `${yIndex}-${xIndex}`;
-
-      gridHash[key] = gridHash[key] || {count: 0, points: [], lonIdx: xIndex, latIdx: yIndex};
-      gridHash[key].count += 1;
-      gridHash[key].points.push(pt);
+      if (
+        !projectPoints ||
+        // when doing screen space agggregation, filter points outside of the viewport range.
+        (xIndex >= 0 && xIndex < numCol && yIndex >= 0 && yIndex < numRow)
```

Comment:
The filter needs to be generalized. What if the aggregator is used for a non-screen grid use case?

Reply:
`projectPoints` argument differentiates, screen space vs non screen space. This aggregator currently being used for both cases.
Updated the header doc for the public function.

Reply:
If there are only two cases, then you should be able to use `viewport.width` and `viewport.height` instead of sending them separately.

Reply:
`width` and `height` are grid dimensions, when doing world space aggregation they refer to bounding box, when doing screen space they refer to viewport.

Reply:
As far as I can tell you are not using `width` and `height` in the world space aggregation case. This goes back to the confusion about how generic this method should be. If you are targeting very specific use cases and not caring about future expansion, why add arguments that you don't need?

Reply:
Ok, bounds checking only needed in screen space (as world space, entire bounding box is mapped to the grid), updated the PR to deduce `width` and `height` from `viewport`.